source/imaer-util/src/main/java/nl/overheid/aerius/util/OptionsMetadataUtil.java

```diff
@@ -0,0 +1,132 @@
+    addBooleanValue(mapToAddTo, Option.WITHOUT_SOURCE_STACKING, !options.isStacking(), addDefaults);
+    if (addDefaults || options.getRoadOPS() != CalculationRoadOPS.DEFAULT) {
+      mapToAddTo.put(Option.OPS_ROAD, options.getRoadOPS().name());
+    }
+    addBooleanValue(mapToAddTo, Option.FORCED_AGGREGATION, options.isForceAggregation(), addDefaults);
+    addBooleanValue(mapToAddTo, Option.USE_RECEPTOR_HEIGHT, options.isUseReceptorHeights(), addDefaults);
+    addConnectSuppliedOptions(options.getConnectSuppliedOptions(), mapToAddTo, addDefaults);
+    addIntValue(mapToAddTo, Option.MONITOR_SRM2_YEAR, options.getMonitorSrm2Year(), addDefaults);
+    opsOptionsToMap(options.getOpsOptions(), mapToAddTo, addDefaults);
+
+    final Map<String, String> returnMap = new LinkedHashMap<>();
+    mapToAddTo.forEach((key, value) -> returnMap.put(key.toKey(), value));
+    return returnMap;
+  }
+
+  private static void addConnectSuppliedOptions(final ConnectSuppliedOptions connectSuppliedOptions, final Map<Option, String> mapToAddTo,
+      final boolean addDefaults) {
+    if (connectSuppliedOptions != null) {
+      addValue(mapToAddTo, Option.CALCULATION_YEAR, connectSuppliedOptions.getCalculationYear(), addDefaults);
+      addValue(mapToAddTo, Option.RECEPTOR_SET, connectSuppliedOptions.getReceptorSetName(), addDefaults);
+    }
+  }
+
+  private static void opsOptionsToMap(final OPSOptions options, final Map<Option, String> mapToAddTo, final boolean addDefaults) {
+    if (options != null) {
+      addBooleanValue(mapToAddTo, Option.OPS_RAW_INPUT, options.isRawInput(), addDefaults);
+      addValue(mapToAddTo, Option.OPS_YEAR, options.getYear(), addDefaults);
+      addValue(mapToAddTo, Option.OPS_COMP_CODE, options.getCompCode(), addDefaults);
+      addValue(mapToAddTo, Option.OPS_MOL_WEIGHT, options.getMolWeight(), addDefaults);
+      addValue(mapToAddTo, Option.OPS_PHASE, options.getPhase(), addDefaults);
+      addValue(mapToAddTo, Option.OPS_LOSS, options.getLoss(), addDefaults);
+      addValue(mapToAddTo, Option.OPS_DIFF_COEFF, options.getDiffCoeff(), addDefaults);
+      addValue(mapToAddTo, Option.OPS_WASHOUT, options.getWashout(), addDefaults);
+      addValue(mapToAddTo, Option.OPS_CONV_RATE, options.getConvRate(), addDefaults);
+      addValue(mapToAddTo, Option.OPS_ROUGHNESS, options.getRoughness(), addDefaults);
+    }
+  }
+
+  private static void addValue(final Map<Option, String> mapToAddTo, final Option key, final Object value, final boolean addDefaults) {
+    // false should be the default value, so only add if value is true, or if defaults should be added anyway.
+    if (value != null) {
+      mapToAddTo.put(key, value.toString());
+    } else if (addDefaults) {
+      mapToAddTo.put(key, "");
+    }
+  }
+
+  private static void addIntValue(final Map<Option, String> mapToAddTo, final Option key, final int value, final boolean addDefaults) {
+    if (addDefaults || value != 0) {
```

Comment:
This seems a bit tricky, because when 0 is an valid value it's not added (Give the only use case here it seems not a problem now, but might be in the future. Maybe add an option to specify a `no_value` and not add if it matches that value.